### PR TITLE
feat: adding default file extension for users

### PR DESCRIPTION
**Pull request by boyney123**
**Description**

When user uses the `new` command they are asked for the `filename`. 

If the user does not provide a file extension the file is created.

This PR will check to see if any file extension has been made, and will add `.yaml` onto the filename if no extension has been provided.

### Why

Just think it's a slightly better developer experience than allowing them to have files without extensions as async API files.

Related issue #114

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -81,17 +81,20 @@ export default class New extends Command {
   async createAsyncapiFile(fileName:string) {
     const defaultAsyncapiFile = await readFile(resolve(__dirname, '../../assets/asyncapi.yaml'), { encoding: 'utf8' });
 
+    const fileNameHasFileExtension = fileName.split('.').pop() !== fileName;
```
Comment (fmvilas) [inline src/commands/new.ts:84]: What about this?

```suggestion
    const fileNameHasFileExtension = fileName.includes('.')
```

Creating an array and removing elements is an expensive operation. I think we can just check for the presence of a `.` char.

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: src/commands/new.ts (hunk 1 above)
Comment (boyney123) [reply src/commands/new.ts:84]: Yeah happy to go with that, would only mean things like `application.random.yaml` would not be valid, although I have no idea who would do this stuff.... pop means we always get the last one regardless of the amount of `.` in the filename.

What do you think? Am I over thinking it :D?

Comment (fmvilas) [reply src/commands/new.ts:84]: But if someone does `application.random.yaml` then it means `application.random` is the name, isn't it?

Comment (fmvilas) [reply src/commands/new.ts:84]: None of our solutions will cover that case, I think. Unless we obtain the extension and check if it's `yml`, `yaml`, or `json` and therefore dicard `random` or any other string 🤔

Comment (boyney123) [reply src/commands/new.ts:84]: > None of our solutions will cover that case, I think. Unless we obtain the extension and check if it's yml, yaml, or json and therefore dicard random or any other string 🤔

Yeah think you are right. Hmmm think let's go for includes and see if we get any issues about it, and in future if we do we can sniff out the extensions we want

**Review by fmvilas — APPROVED** [review]

**Comment by boyney123** [comment]
Thanks for the review @fmvilas , if you are cool @Souvikns maybe it's good to go?

**Comment by Souvikns** [comment]
Yeah I think this is good to go, I am merging it.

**Review by Souvikns — APPROVED** [review]

**Pull request merged**